clients/cobol-lsp-vscode-extension/src/task/CustomBuildTaskTerminal.ts

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2022 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+
+import { IProfileLoaded } from "@zowe/imperative";
+import { ZoweExplorerApi } from "@zowe/zowe-explorer-api";
+import * as fs from "fs";
+import * as path from "path";
+import * as vscode from "vscode";
+import { SettingsService } from "../services/Settings";
+import { COMPILE_JCL_HEADER, COMPILE_JCL_MIDDLE, COMPILE_JCL_TAIL } from "./script/compile";
+
+export class CustomBuildTaskTerminal implements vscode.Pseudoterminal {
+    private ERROR_REGEX = "^\\s+([0-9]*)\\s+IGY[A-Z]{2}\\d{4}-([A-Z])\\s+(.*)$";
+    private writeEmitter = new vscode.EventEmitter<string>();
+    onDidWrite: vscode.Event<string> = this.writeEmitter.event;
+    private closeEmitter = new vscode.EventEmitter<number>();
+    onDidClose?: vscode.Event<number> = this.closeEmitter.event;
+
+    constructor(private documentText: vscode.TextDocument, private definition: vscode.TaskDefinition) {
+    }
+
+    public open(_initialDimensions: vscode.TerminalDimensions | undefined): void {
+        this.doCompile();
+    }
+
+    public close(): void {
+    }
+
+    private async doCompile(): Promise<void> {
+        const zoweExplorerApi: ZoweExplorerApi.IApiRegisterClient = vscode.extensions.getExtension("Zowe.vscode-extension-for-zowe")?.exports;
+        let loadedProfile: IProfileLoaded;
+        if (!zoweExplorerApi) {
+            return this.handleError({ message: "Please install zowe Explorer" });
+        }
+        if (!this.definition.jobCard) {
+            return this.handleError({ message: "jobCard is mandatory. customize task and provide these attributed in task.json" });
```

Comment:
Can we show the name of the task with a link to the task.json in the message?
```suggestion
            return this.handleError({ message: "jobCard missing. Update [{$task name}](link to the task.json file) task configuration.
```

Reply:
It seems complicated to perform on the terminal as against the vscode error window.

Reply:
The vscode configure task adds a new element to the tasks array and doesn't simply take us to the tasks.json

Reply:
Got it, tahnk you for the explanation. In that case I would suggest
```suggestion
            return this.handleError({ message: "jobCard is missing. Update the task configuration in task.json ." });
```